Sources/SmithyReadWrite/DocumentReadingClosure.swift

```diff
@@ -0,0 +1,14 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.Data
+
+public typealias DocumentReadingClosure<T, Reader> = (Data, ReadingClosure<T, Reader>) throws -> T
```

Comment:
This is the closure type that we provide to `DeserializeMiddleware` to read the HTTP response data to an output model.